test/tstCompileOnlyTypeRequirements.cpp

```diff
@@ -91,3 +91,41 @@ void check_bounding_volume_and_predicate_geometry_type_requirements()
              KOKKOS_LAMBDA(NearestPredicate, auto){});
 #endif
 }
+
+namespace Test
+{
+
+// clang-format off
+struct FakePrimitiveGeometry {};
+
+template <int DIM>
+KOKKOS_FUNCTION void expand(ArborX::Box<DIM> &, FakePrimitiveGeometry) {}
+KOKKOS_FUNCTION auto returnCentroid(FakePrimitiveGeometry) { return ArborX::Point<3>{}; }
+// clang-format on
+
+} // namespace Test
+
+template <>
+struct ArborX::GeometryTraits::dimension<Test::FakePrimitiveGeometry>
+{
+  static constexpr int value = 3;
+};
+template <>
+struct ArborX::GeometryTraits::coordinate_type<Test::FakePrimitiveGeometry>
+{
+  using type = float;
+};
+
+// Compile-only
+void check_hierarchy_for_custom_types()
+{
+  using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+  using MemorySpace = ExecutionSpace::memory_space;
+  using Tree = ArborX::BoundingVolumeHierarchy<
+      MemorySpace, Test::FakePrimitiveGeometry,
+      ArborX::Experimental::DefaultIndexableGetter>;
```

Comment:
Why did you decide to spell out the Tree type here rather than using CTAD below?

Reply:
No good reason.

Reply:
Then I would prefer using CTAD.